tee/test/generictest.go

```diff
@@ -0,0 +1,268 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"perun.network/go-perun/pkg/test"
+
+	"github.com/perun-network/erdstall/eth"
+	"github.com/perun-network/erdstall/tee"
+)
+
+type clientAction = func(ctx context.Context, bal *tee.BalanceProof) error
+
+type Client interface {
```

Comment:
We discussed to move `tee/test.NewRandomAddress` into package `eth` to break the dependency and then to delete this interface and use the normal `client/test.Client` in the `GenericEnclaveTest`.